cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -419,11 +430,15 @@ public ApplicationMeta getLatest(ApplicationReference appReference) throws IOExc
   }
 
   public List<ApplicationMeta> getAllAppVersions(ApplicationReference appRef) throws IOException {
-    return scanWithRange(
-      getNamespaceAndApplicationRange(appRef),
-      ApplicationMeta.class,
-      getApplicationSpecificationTable(),
-      StoreDefinition.AppMetadataStore.APPLICATION_DATA_FIELD);
```

Comment:
`ApplicationMeta` cannot be fully deserialized from `APPLICATION_DATA_FIELD`. Instead, call decodeRow